src/transaction/transaction.rs

```diff
@@ -278,6 +343,49 @@ impl TwoPhaseCommitter {
         }
     }
 
+    async fn pessimistic_commit(mut self, for_update_ts: u64) -> Result<()> {
```

Comment:
If possible, it would be nice for these two methods to share code with the optimistic versions.